src/fsharp/PrettyNaming.fs

```diff
@@ -484,30 +484,36 @@ module public Microsoft.FSharp.Compiler.PrettyNaming
      
     let [<Literal>] private mangledGenericTypeNameSym = '`'
 
-    let IsMangledGenericName (n:string) = 
-        n.IndexOf mangledGenericTypeNameSym <> -1 &&
+    let IsMangledGenericNameAndPos (n:string) =
         (* check what comes after the symbol is a number *)
-        let m = n.LastIndexOf mangledGenericTypeNameSym
-        let mutable res = m < n.Length - 1
-        for i = m + 1 to n.Length - 1 do
-            res <- res && n.[i] >= '0' && n.[i] <= '9'
-        res
+        let pos = n.LastIndexOf mangledGenericTypeNameSym
+        if pos = -1 then None else
+        let mutable res = pos < n.Length - 1
+        let mutable i = pos + 1
+        while res && i < n.Length do
+            let char = n.[i]
+            if not (char >= '0' && char <= '9') then
+                res <- false
+            i <- i + 1
+        if res then
+            Some pos
```

Comment:
You can use `ValueSome` from `illib.fs` if you like.  There is also `ValueSome` in the F# library now, I suppose we should remove the one in illib.fs (or better, make it conditional since it will still be used by FCS which depends on lower FSHarp.Core)